File: app/src/main/java/com/example/task2/ui/FragmentAdapter.java
package com.example.task2.ui;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.example.task2.ui.benchmark.BenchmarksFragment;

public class FragmentAdapter extends FragmentStateAdapter {

    public FragmentAdapter(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle) {
        super(fragmentManager, lifecycle);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return BenchmarksFragment.newInstance(position);
    }

    @Override
    public int getItemCount() {
        return 2;
    }
}

File: app/src/main/java/com/example/task2/ui/FragmentAdapter.kt
package com.example.task2.ui

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.task2.ui.benchmark.BenchmarksFragment

class FragmentAdapter(
    fragmentManager: FragmentManager, lifecycle: Lifecycle
) : FragmentStateAdapter(fragmentManager, lifecycle) {

    override fun createFragment(position: Int): Fragment = BenchmarksFragment.newInstance(position)

    override fun getItemCount(): Int = 2
}
